### PR TITLE
[Feature][Connector-V2][SFTP] Add SFTP file connector (read & write)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

#3018 SFTP file connector

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: docs/en/connector-v2/sink/SftpFile.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+| username                         | string  | yes      | -                                                         |
+| password                         | string  | yes      | -                                                         |
+| path                             | string  | yes      | -                                                         |
+| file_name_expression             | string  | no       | "${transactionId}"                                        |
+| file_format                      | string  | no       | "text"                                                    |
+| filename_time_format             | string  | no       | "yyyy.MM.dd"                                              |
+| field_delimiter                  | string  | no       | '\001'                                                    |
+| row_delimiter                    | string  | no       | "\n"                                                      |
+| partition_by                     | array   | no       | -                                                         |
+| partition_dir_expression         | string  | no       | "\${k0}=\${v0}\/\${k1}=\${v1}\/...\/\${kn}=\${vn}\/"      |
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/SftpFile.md]: "${k0}=${v0}/${k1}=${v1}/.../${kn}=${vn}/"   is right.

File: docs/en/connector-v2/sink/SftpFile.md [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+Only support `true` now.
+
+### save_mode [string]
+
+Storage mode, currently supports `overwrite`. This means we will delete the old file when a new file have a same name with it.
+
+If `is_enable_transaction` is `true`, Basically, we won't encounter the same file name. Because we will add the transaction id to file name.
+
+For the specific meaning of each mode, see [save-modes](https://spark.apache.org/docs/latest/sql-programming-guide.html#save-modes)
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/SftpFile.md]: I think we need add document to introduce seatunnel `save_mode` not Spark or Flink.

File: docs/en/connector-v2/sink/SftpFile.md [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+## Description
+
+Output data to Sftp .
+
+## Key features
+
+- [x] [exactly-once](../../concept/connector-v2-features.md)
+- [ ] [schema projection](../../concept/connector-v2-features.md)
+
```
Comment (EricJoy2048) [inline docs/en/connector-v2/sink/SftpFile.md:22]: Please add :

```
- [x] file format
    - [x] text
    - [x] csv
    - [x] parquet
    - [x] orc
    - [x] json
```

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Comment by EricJoy2048** [comment]
@CalvinKirs  PTAL

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**